src/s3.js

```diff
@@ -1,9 +1,14 @@
 var AWS = require('aws-sdk')
 var denodeify = require('denodeify')
 
-module.exports = function(cfg) {
-    var credentials = new AWS.SharedIniFileCredentials(cfg.aws_profile ? {profile: cfg.aws_profile} : {});
-    AWS.config.credentials = credentials;
+module.exports = async function(cfg) {
+    AWS.config.credentials = new AWS.EC2MetadataCredentials({
+        httpOptions: { timeout: 5000 }, // 5 second timeout
+        maxRetries: 10, // retry 10 times
+        retryDelayOptions: { base: 200 }, // see AWS.Config for information
+        logger: console // see AWS.Config for information
+      });
```

Comment:
I think we'll need a credential provider chain with two elements:
  1. Profile credentials
  2. EC2 credentials

The former is necessary to enable this library to be used locally, as we'll never have EC2 credentials.